node/impl/full.go

```diff
@@ -37,4 +40,40 @@ func (n *FullNodeAPI) CreateBackup(ctx context.Context, fpath string) error {
 	return backup(n.DS, fpath)
 }
 
+func (n *FullNodeAPI) NodeStatus(ctx context.Context) (status api.NodeStatus, err error) {
+	curTs, err := n.ChainHead(ctx)
+	if err != nil {
+		return status, err
+	}
+
+	status.SyncStatus.Epoch = uint64(curTs.Height())
+	timestamp := time.Unix(int64(curTs.MinTimestamp()), 0)
+	delta := time.Since(timestamp).Seconds()
+	status.SyncStatus.Behind = uint64(delta / 30)
+
+	pis, err := n.NetPeers(ctx)
+	if err != nil {
+		return status, err
+	}
+
+	scores, err := n.NetPubsubScores(ctx)
+	if err != nil {
+		return status, err
+	}
+
+	scoreMap := make(map[peer.ID]float64)
+	for _, score := range scores {
+		scoreMap[score.ID] = score.Score.Score
```

Comment:
This map could be large, and it looks like too much detail for most use cases. Can we put it behind a flag?

Reply:
Hrm, maybe; it's kind of important however, at least for the logical health checks.

Reply:
also, how large is large? It won't be more than 1k in practice.

Reply:
we don't report specific peers, just the count of peers above the threshold.

Reply:
Oh, I see. It's a bit confusing to follow due to the mixture of assignments between local vars and result struct fields.

Reply:
done.